app/controllers/offshore/YouHaveNotIncludedTheTaxYearController.scala

```diff
@@ -0,0 +1,84 @@
+import models.{Mode, UserAnswers, TaxYearStarting}
+import navigation.OffshoreNavigator
+import pages.YouHaveNotIncludedTheTaxYearPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.SessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.offshore.YouHaveNotIncludedTheTaxYearView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class YouHaveNotIncludedTheTaxYearController @Inject()(
+                                        override val messagesApi: MessagesApi,
+                                        sessionService: SessionService,
+                                        navigator: OffshoreNavigator,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: YouHaveNotIncludedTheTaxYearFormProvider,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        view: YouHaveNotIncludedTheTaxYearView
+                                    )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+
+      val (firstYear, missingYear, lastYear) = getMissingYearData(request.userAnswers)
+
+      val preparedForm = request.userAnswers.get(YouHaveNotIncludedTheTaxYearPage) match {
+        case None => formProvider(missingYear)
+        case Some(value) => formProvider(missingYear).fill(value)
+      }
+
+      Ok(view(preparedForm, mode, firstYear, missingYear, lastYear))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      val (firstYear, missingYear, lastYear) = getMissingYearData(request.userAnswers)
+
+      formProvider(missingYear).bindFromRequest().fold(  
+        formWithErrors => 
+          Future.successful(BadRequest(view(formWithErrors, mode, firstYear, missingYear, lastYear))),
+
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(YouHaveNotIncludedTheTaxYearPage, value))
+            _              <- sessionService.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(YouHaveNotIncludedTheTaxYearPage, mode, updatedAnswers))
+      )
+  }
+
+  private def getMissingYearData(userAnswers: UserAnswers): (String, String, String) = {
+    val taxYearStartingList = userAnswers.inverselySortedOffshoreTaxYears.toList.flatten
+    val missingYears = TaxYearStarting.findMissingYears(taxYearStartingList)
+    val missingYear = if (missingYears.nonEmpty) missingYears(0).toString else "0"
```

Comment:
So the best way to handle this is to use `.headOption` which gives you Some(missingYear) or None depending on if there is a populated list.

I think in this controller we want to handle the situation where we're on this page incorrectly, and we want to redirect the user to the right place if we don't actually have any missing years.

So if there are no missing years, I think we redirect them to the Country page because they shouldn't be on this page